frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue

```diff
@@ -31,14 +35,60 @@ const valueProperty = computed({
 
   set(newValue) {
     const selectedOption: OneOfAnyOfSelectionOption | undefined = newValue;
+    const path = pathToString(props.absolutePath);
 
     if (selectedOption) {
-      const path = pathToString(props.absolutePath);
       useSessionStore().currentSelectedOneOfOptions.set(path, selectedOption);
+      applySchemaConstantsOnDataBasedOnSelection(props.absolutePath, selectedOption);
+      emit('update_tree');
+    } else {
+      useSessionStore().currentSelectedOneOfOptions.delete(path);
       emit('update_tree');
     }
   },
 });
+
+function applySchemaConstantsOnDataBasedOnSelection(
+  path: Path,
+  selectedOneOfOption: OneOfAnyOfSelectionOption
+) {
+  // Based on the user selection, a schema is created by merging the base schema
+  // of the property with the selected oneOf.
+  // If the merged schema contains constants, those are applied to the data.
+  // Note that it might seem logical to compute this merged schema only once
+  // (e.g. when oneOf is selected), store it and re-use it later.
+  // This, however, is not done because the effective merged schema could change
+  // even without a new oneOf selection by the user, due to JSON schema features
+  // such as if and else.
+
+  // TODO: maybe this function should instead of normal schema make use of the effective schema
+  const schemaAtPath = useSessionStore().schemaAtPath(path);
```

Comment:
yes it should use the effective schema

Reply:
In fact, we should probably make a warning to `schemaAtPath` that `effectiveSchemaAtPath` should be used in most cases

Reply:
fixed